### ISSUE TITLE
Caja's SEARCH Location/Criteria Has HUGE Menus/Buttons

### ISSUE THREAD

**Issue by UbuntuMetal** [mate-desktop/caja#1689]
#### Expected behaviour
Caja has normal size **Location buttons** for pull-down menus in **Search results** under Search bar.

#### Actual behaviour
After a recent update, Caja has **massive** location buttons in the Search criteria area once you perform a search.

#### Steps to reproduce the behaviour
Perform a search, and the location buttons/menus appear as you see in the screenshots. The only thing I can think of that _may_ have contributed to this is I had a small update, and I noticed that **mate-menus** was being updated (but I can't see how this would have affected Caja). This behaviour started after the next reboot, and does not affect any other apps.

While this isn't an emergency, and Search functions normally otherwise, it is a tad distracting. If anyone knows of a workaround, like a config file to edit, or some other hack, please let me know.

#### MATE general version
1.26.0

#### Package version
caja 1.26.0-1ubuntu1

#### Linux Distribution
Ubuntu 22.04 with MATE & KDE

#### Link to bugreport of your Distribution (requirement)
caja:
  Installed: 1.26.0-1ubuntu1
  Candidate: 1.26.0-1ubuntu1
  Version table:
 *** 1.26.0-1ubuntu1 500
        500 http://archive.ubuntu.com/ubuntu jammy/universe amd64 Packages
        100 /var/lib/dpkg/status

Maintainer: Ubuntu Developers <ubuntu-devel-discuss@lists.ubuntu.com>
Original-Maintainer: Debian+Ubuntu MATE Packaging Team <debian-mate@lists.debian.org>
Bugs: https://bugs.launchpad.net/ubuntu/+filebug

![Caja Search Criteria](https://user-images.githubusercontent.com/41879371/207998251-631cdac2-fed1-49cd-bec1-dea69baec6c2.png)
![Caja Search Weirdness](https://user-images.githubusercontent.com/41879371/207998270-7909728d-03c1-4c9e-96b0-10a7e077fa26.png)

**Comment by lukefromdc** [comment]
What theme are you running?This is controlled by your GTK theme unless hardcoded and I don't think we ever hardcoded anything on that button

**Comment by UbuntuMetal** [comment]
> What theme are you running

That's my own one, based on Ambiant MATE Dark. This weirdness **only** affected Caja, in both MATE and KDE. While I found the way to rectify this, it's worth noting I couldn't replicate this behaviour anywhere else. And as outlined extensively above, it had to do with Caja not being able to handle a larger icon size than it expected, which wasn't a problem for any other program, including 3 other file-managers: Nautilus, Nemo and Thunar. So since it was fixed by creating multiple icons of different sizes, it could hardly have anything to do with a theme.

And in case you meant icon theme (which is also my creation), then I think this is irrelevant too, since it was a custom drive icon that broke Caja.

**Comment by lukefromdc** [comment]
There were a LOT of oversize icon problems in GTK 3 apps back in the mid 2010's, as both  GTK 3 and 3ed party icon themes matured most of them (but by your experience not all of them) have disappeared.

Icon themes should NOT affect the size of widgets containing them, an oversize icon should be rescaled with ONE exception: a directory of icons designated in the icon theme index as non-resizable will NOT be rescaled. When I learned about that I never saw this oversize icon problem again

 GTK theme controls the padding around icons and anything else needing space around itself.

**Comment by crojack** [comment]
I don't understand why icon theme creators still use png icons instead of scalable (svg) ones. With svg icons you don't need to have so many directories in your icon theme and the index.theme is so much easier to write. And there is no oversizing if you set minimum icon size to 16 pixels.

**Comment by lukefromdc** [comment]
Historically, .svg icons also had a disadvantage: they were significantly heavier. Less of an issue now when phones are creeping up on early 2000's workstation power, but back in 2008 the first bug report I ever posted was for the then-current UbuntuStudio icon theme causing a big directory full of audio files to take forever to open-and have rendering issues on a Pentium 4/1.8 GHZ machine.  I downscaled that icon to a smaller resolution and it worked well on that and similar machines, so some of it was just too much resolution too early on a beautiful icon set I still use a descendent of I now maintain myself. 

The next UbuntuStudio icon theme was for XFCE, which they transitioned to to avoid gnome-shell in 2011. It was all .png icons and much lighter, but then as now, a missing small icon can cause some GTK widgets to render the larger icon fullsize.

**Comment by crojack** [comment]
> Historically, .svg icons also had a disadvantage: they were significantly heavier. Less of an issue now when phones are creeping up on early 2000's workstation power, but back in 2008 the first bug report I ever posted was for the then-current UbuntuStudio icon theme causing a big directory full of audio files to take forever to open-and have rendering issues on a Pentium 4/1.8 GHZ machine. I downscaled that icon to a smaller resolution and it worked well on that and similar machines, so some of it was just too much resolution too early on a beautiful icon set I still use a descendent of I now maintain myself. The next UbuntuStudio icon theme was for XFCE, which they transitioned to to avoid gnome-shell in 2011. It was all .png icons and much lighter, but then as now, a missing small icon can cause some GTK widgets to render the larger icon fullsize.

Again, I don't understand the "missing small icons issue" when we have "Inherits" entry in index.theme.

By the way, do you know how to make Caja side panel icons bigger? I am visually impaired and although I use Mate as my desktop I am forced to use Thunar where I can scale side panel icons up to 48 px.

**Comment by lukefromdc** [comment]
Right-click on the left panel (assuming you are using a mate-panel set to render on the left), and select "Properties" from the context menu. You will see a "size" spin box item. Substantially increase the size and the icons should scale with it. It they don't, try another icon theme

**Comment by crojack** [comment]
I know how to resize mate-panel icons. What I don't know is how to make Caja side panel icons bigger. It seems they are hardcoded to 16 pixels.

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by lukefromdc** [comment]
No problem, I misread this. Caja does not have a setting to change side panel icon size, at least not that I've ever seen. That said there are workarounds

If you run compiz, there is a "zoom" plugin that is a screen magnifier and will show everything in the zoomed area double-size.

**Comment by crojack** [comment]
I am using Thunar file manager now. It can scale side panel icons up to 1024 pixels, but I'd like to have that option in Caja because Caja handles Mate desktop and it's impossible to open any desktop folder with other file manager than Caja.

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by lukefromdc** [comment]
I don't have the time to code that at the moment, busy with the Wayland porting effort (we will support both Xorg and wayland). If anyone else wants to work on this I'll be glad to review it though

**Comment by crojack** [comment]
Will we get proper scaling in Mate with Wayland?

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by lukefromdc** [comment]
In wayland itself you can set fractional scaling and (at least in wayfire) it just plain works. This is one of the things that's a lot easier in wayland than in x11.
